Incorrect Python version used in Build & Publish workflow

When setting up the build job, Python version is hardcoded to `3.12`.
Should use `matrix.python`.

It works well for Windows with Python 3.12, but fails for Windows with Python {3.10, 3.11, 3.13}.
The error message reads:
`assert _sre.MAGIC == MAGIC, "SRE module mismatch"\nAssertionError: SRE module mismatch\`

Probably related to https://github.com/astral-sh/uv/issues/17814

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -24,17 +24,16 @@ jobs:
       - uses: actions/checkout@v6
         with:
           fetch-depth: 0
-      
+
       - name: Install uv
         uses: astral-sh/setup-uv@v7
-
-      - name: Set up Python
-        run: uv python install 3.12
+        with:
+          python-version: ${{ matrix.python }}
 
       - name: Install cibuildwheel
         run: |
-         uv venv
-         uv pip install -U pip cibuildwheel twine
+          uv venv
+          uv pip install -U pip cibuildwheel twine
 
       - name: Build wheels
         run: uv run cibuildwheel --output-dir wheelhouse

